### ISSUE TITLE
Combined induced charge assigned to a grabbed balloon

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#336]
Induced charge amount "a little bit" becomes "a lot" when balloons overlap.

<img width="877" alt="base-induced-charge-a-little" src="https://user-images.githubusercontent.com/1902116/33452403-ae2b0b7e-d5ec-11e7-9fd3-a97c7a11d00d.png">
<img width="871" alt="base-indiced-charge-a-lot" src="https://user-images.githubusercontent.com/1902116/33452411-b29d01a8-d5ec-11e7-8cab-732bfaf0fd25.png">

To address this issue, I have designed a proximity threshold and some additional components to the rubbing alerts. These description designs are outlined by alert in [Two-balloon experiments](https://docs.google.com/document/d/1LrOWT4RZiS4emrumW479POK--zWveWiylZMd7unNyi0/edit#heading=h.bnj53lyu3roi) design document. For the above scenario see specifically [section 3A](https://docs.google.com/document/d/1LrOWT4RZiS4emrumW479POK--zWveWiylZMd7unNyi0/edit#heading=h.bnj53lyu3roi) in the design document.

With this change, the second screen shot above would have this alert:
"At wall[[, next to yellow balloon]]. No transfer of charge. Negative charges in [[wall]] move away from [[each balloon]] a [[little bit]]. Positive charges do not move."

@jessegreenberg, do you have any concerns about this Proximity Threshold approach? I have sketched it out for several alerts that contain "location" information.

**Comment by jessegreenberg** [comment]
> Induced charge amount "a little bit" becomes "a lot" when balloons overlap.

This is actually physically correct, right? When the balloons overlap, the charge description *should* be "a lot".

I have no issue with using [[each balloon]] when balloons overlap.

**Comment by terracoda** [comment]
@jessegreenberg, the issue is that, the user has the green balloon grabbed, but gets no information (yet) of the whereabouts of the ungrabbed balloon. The amount of deflection is a combined effect.

I was thinking this issue could be solved in either of two ways with  a proximity description and adjustments to balloon and charge descriptions. For example, I considered both these approaches:
1. "At wall[[, next to yellow balloon]]. No transfer of charge. Negative charges in [[wall]] move away from [[each balloon]] a [[little bit]]. Positive charges do not move."
OR
2. "At wall[[, next to yellow balloon]]. No transfer of charge. Negative charges in [[wall]] move away from [[balloons]] a [[a lot]]. Positive charges do not move."

I went with **number 1** because I thought it was easier, but it actually requires a change to both the balloon description and the charge description.

Perhaps, **number 2** is more efficient and more accurate according to the model. It requires the use of "balloons", but the description of deflection can be left as-is to the model.

@jessegreenberg, do you like option 2 better?

**Comment by jessegreenberg** [comment]
I understand now @terracoda thanks for clarifying. I personally prefer option 2 because I think it is more physically accurate.

**Comment by terracoda** [comment]
@jessegreenberg, as I said in slack, I'll have to update the design documentation to change the perspective to a combined two-balloon effect on induced change rather than an individualized balloon effect on induced charge. This hopefully will be way more efficient.

Just let me know before you get to this.

**Comment by jessegreenberg** [comment]
@terracoda since approach/return alerts are done I could work on this soon.

**Comment by terracoda** [comment]
@jessegreenberg, I have updated the section in the design document that covers the Show all charges view. The alerts are much shorter. 

I am still updating the Show charge differences view. There is no induced charge effect for that view, so the situation is a little different.

The primary view, "Show all charges" is way better now.

**Comment by terracoda** [comment]
@jessegreenberg, I have updated the design document for All charge views, of course, Show no charges is not affected at all.

I think Proximity Phrase for against the wall are ready for implementation.

**Comment by terracoda** [comment]
I am unassigning myself.

**Comment by terracoda** [comment]
@jessegreenberg, due to this change, I have to review the proximity phrases in the other sections. I like the describing proximity at the wall because of the combined induced charge effect. 

I am now not sure if we need proximity phrases everywhere.

**Comment by terracoda** [comment]
@jessegreenberg, I think I was a little unclear in my previous comment posted on Dec 19th.

The Proximity Phrases for against the wall are indeed ready for implementation with one remaining question. The examples in the design document for _Show all charges_, _Show no charges_, and _Show charge differences_ were all updated.

Just noting here that the descriptions are now much shorter in the main view, _Show all charges_, which is a huge benefit, and the _Show no charges_ is not affected at all.

For the _Show charges differences_ view the descriptions are longer, however; this view is not the primary view, so I think this is acceptable.

**One remaining Question for _Show charge differences_**
In the two cases where the balloons could potentially have the same amount or similarly described amounts of negative charge:
- case 1: they each have "showing a few negative charges"
- case 2: they each have "showing several negative charges" 

**Question:** Is it pedagogically clear to describe the sum of negative charges showing on the balloons like this:
- case 1: At [[wall]][[, next to green balloon]]. [[Balloons]] have [[negative]] net charge, **showing [[several]] negative charges**. Wall has [[zero]] net charge, showing [[no]] charges. 
- case 2: At [[wall]][[, next to green balloon]]. [[Balloons]] have [[negative]] net charge, **showing [[many]] negative charges**. Wall has [[zero]] net charge, showing [[no]] charges. 

If it is not pedagogically clear, then we would have to have two special cases that describe each balloon as showing the same amount of charge like this:

- Special case 1: At [[wall]][[, next to green balloon]]. [[Balloons]] have [[negative]] net charge, **[[each]] showing [[a few]] negative charges**. Wall has [[zero]] net charge, showing [[no]] charges.
- Special case 2: At [[wall]][[, next to green balloon]]. [[Balloons]] have [[negative]] net charge, **[[each]] showing [[several]] negative charges**. Wall has [[zero]] net charge, showing [[no]] charges.

Will add screen shots for each case in a follow-up comment.
Also, assigning to @emily-phet for comment.

**Comment by terracoda** [comment]
**To contrast visually with Show all charges** 
In Show all charges view, the **sum of charges** will be used to describe the amount of deflection in the wall:
Case 1 with Show all Charges view: Each balloon has a few negative charges totaling "several charges", so the description of the deflection is described as "Negative charges in wall move away from balloons a lot." 
<img width="525" alt="show-all-2-balloons-few-charges" src="https://user-images.githubusercontent.com/1902116/34615551-df6e3334-f20f-11e7-9fb1-d418de776669.png">

Case 2 with Show all Charges view: When each balloon has several negative charges totaling "many charges", the description of the deflection is described as "Negative charges in wall move away from balloons quite a lot." 
<img width="553" alt="balloons-several-charges-at-wall" src="https://user-images.githubusercontent.com/1902116/34615151-607fea50-f20e-11e7-8f9c-82e987f9dc72.png">

**Comment by terracoda** [comment]
Decision: keep it simple and consistent when possible.

In Jan 5th A11y meeting, EM suggested that we just go with no special cases (no shorter combined descriptions when objects have similar negative net charge) in Show charge differences. 

The only special case now in Show charges differences is when both balloons have zero net charge. Please see design document.

I still am still reviewing/updating the Grab Alerts.

**Comment by terracoda** [comment]
@emily-phet, the decision to not use a combined description in the case of dragging in Show charges differences may affect my earlier design of a combined approach in the Grab Alert for Show all charges.

@emily-phet, can we discuss?

**Comment by emily-phet** [comment]
Sure - let's talk about this on Tuesday in our meeting.

**Comment by terracoda** [comment]
@jessegreenberg, you said 
> @terracoda since approach/return alerts are done I could work on this soon.

Yes, let's do that. Sorry I missed your https://github.com/phetsims/balloons-and-static-electricity/issues/336#issuecomment-351872282

**Comment by terracoda** [comment]
@jessegreenberg, Approach & Return alerts are great!

@emily-phet and I discussed the Jan 5th decision, and since I took a combined approach in other views and in other alerts, i.e., the Grab Alert, the special case, "[[each]] showing [[x amount]] of negative charges", presented above for Show all charges seems aligned with other special cases.

I'd like to keep these special cases that use "each" if it is not too onerous. Let's discuss more if you think these special cases cause a big slow down on your end.

**Comment by terracoda** [comment]
The "each" string is used in Wall Rubbing for show charge differences, the Grab Alert, and in the Scene Summary.

@jessegreenberg and discussed (Jan 30), where this string is used and all cases seemed quite doable.

**Comment by terracoda** [comment]
@jessegreenberg, I am un-assigning myself. Combined induced charge descriptions and the special cases for all views is laid out in [Section 3A of Two-balloons Scenarios](https://docs.google.com/document/d/1LrOWT4RZiS4emrumW479POK--zWveWiylZMd7unNyi0/edit#heading=h.8p3z1ub0ptku)

**Comment by jessegreenberg** [comment]
@terracoda this was done in the above commits. Thank you so much for the detailed examples in the design document, that really helped. I still need to do the "Show charge differences case", Ill work on that against this issue as well.

**Comment by terracoda** [comment]
Thanks, @jessegreenberg!

**Comment by terracoda** [comment]
@jessegreenberg, I found two issues while testing combiend induced charge and next to phrases for releases and dragging. 

Show charges differences will have next phrases, but it won't be affected by induced charge descriptions.

**Comment by terracoda** [comment]
@jessegreenberg, I think you just implemented the combined "balloons" phrasing while I was testing.

It's working while dragging a balloon, but the string also needs updating in the Grab Alert and the Release alert.

You might be right in the middle of making these changes.

**Comment by jessegreenberg** [comment]
The wall rubbing and balloon grabbing alerts are up to date, but the release alerts need to be updated to reflect latest.

**Comment by jessegreenberg** [comment]
Combined release alert is used when adjacent balloons are both inducing charge on release. After reviewing comments i *think* that is it for this issue, but a fair amount was discussed. i am going to close, but if more things crop up we can create individual issues. Closing.
